### PR TITLE
useFetchClient hook replacing axiosInstance

### PR DESCRIPTION
**Pull request by simotae14**
## Create a common Axios Factory
The idea is to create a simple axios Factory to wrap axios and the admin fetch client that know more about the admin logic (ex: getting auth token).

Suggestions:

- exposing the Admin APIs & tools to build things with Strapi either with the DS or `@strapi/admin` directly. And create this new wrapper in the admin 1st and keeping it private until we expose those a clean API surface in v5.
- differentiating a generic axios wrapper & the admin fetch client that knows more about the admin logic (like getting the auth token)
- instantiate this once in the Strapi App and expose it as a common “service” through the App instance. Hence just create a hook that will get the current instance from the App and return it to be used.
- It shouldn’t be axios dependant, therefore, the name shouldn’t be axiosInstance, because if we change the request library underneath it some day, we would need to replace the name again in all their uses

### Why is it needed?

Inside the Strapi project we have a lot of different implementations of the same code to use axios and its axiosInstance.

For example the axiosInstance is defined in:

- *****admin*****
- ***users-permissions***
- ******upload******
- ****i18n****
- *content-type-builder*

### How to test it?

You need to use the new construct inside every axiosInstance call

### CODE REVIEW

File: packages/core/admin/admin/src/pages/SettingsPage/pages/ApiTokens/EditView/index.js
Comment (simotae14) [inline packages/core/admin/admin/src/pages/SettingsPage/pages/ApiTokens/EditView/index.js:63]: This is an example how we use the hook, we need to remove it before merging our code

File: packages/core/admin/admin/src/StrapiApp.js
Comment (gu-stav) [inline packages/core/admin/admin/src/StrapiApp.js:26]: I think this is not a good location. Can we put this into a `utils` folder?

File: packages/core/admin/admin/src/core/fetch-client/index.js
Comment (gu-stav) [inline packages/core/admin/admin/src/core/fetch-client/index.js:38]: I'd like to keep this interface as small and axios unspecific as possible. What do you think about exposing only the 4 HTTP methods?

```suggestion
export default {
  get: instance.get,
  post: instance.post,
  put: instance.put,
  delete: instance.delete
};
```

This would avoid some ambiguity we have at the moment, because axios allows to run `instance({ method: 'get' })` and `instance.get()`.

File: packages/core/admin/admin/src/pages/SettingsPage/pages/ApiTokens/EditView/index.js
Comment (gu-stav) [inline packages/core/admin/admin/src/pages/SettingsPage/pages/ApiTokens/EditView/index.js:117]: I'm wondering if that would be a good opportunity to move the `/admin/` prefix into `fetchClient`, because it is always required and it would be quite beneficial if we could make this configurable in one place.

File: packages/core/admin/admin/src/StrapiApp.js
Comment (simotae14) [reply packages/core/admin/admin/src/StrapiApp.js]: ok, I agree 😃 Done

File: packages/core/admin/admin/src/pages/SettingsPage/pages/ApiTokens/EditView/index.js
Comment (simotae14) [reply packages/core/admin/admin/src/pages/SettingsPage/pages/ApiTokens/EditView/index.js:117]: do you mean directly inside the baseURL? because I saw inside the code different places where the url passed as argument is a variable, maybe it is better to leave it like this, just because in the future could be beneficial if we want to  change something

File: packages/core/admin/admin/src/core/fetch-client/index.js
Comment (simotae14) [reply packages/core/admin/admin/src/core/fetch-client/index.js]: ok I have modified the code and right now I am exposing these methods:
```
export default {
  getClient: instance.get,
  putClient: instance.put,
  postClient: instance.post,
  deleteClient: instance.delete,
};
```

do you think it is ok? what do you think about the names of the different methods I have used?

Comment (gu-stav) [reply packages/core/admin/admin/src/core/fetch-client/index.js]: I'm not much of a fan of these names. These are not clients, but function calls and I'd prefer plain HTTP verbs. Wdyt?

File: packages/core/admin/admin/src/pages/SettingsPage/pages/ApiTokens/EditView/index.js
Comment (simotae14) [reply packages/core/admin/admin/src/pages/SettingsPage/pages/ApiTokens/EditView/index.js]: For example:
-  packages/core/admin/admin/src/content-manager/components/CollectionTypeFormWrapper/index.js line 140
`const { data } = await axiosInstance.get(requestURL, { cancelToken: source.token });` and requestUrl is coming from
`  const requestURL = useMemo(() => {
    if (isCreatingEntry && !origin) {
      return null;
    }

    return getRequestUrl(`collection-types/${slug}/${origin || id}`);
  }, [slug, id, isCreatingEntry, origin]);`

Comment (simotae14) [reply packages/core/admin/admin/src/pages/SettingsPage/pages/ApiTokens/EditView/index.js]: we have found the getRequestUrl which has a lot of different implementations inside the repo that gave us a similar logic (`packages/core/admin/admin/src/utils/getRequestUrl.js` this one is the correct one for us).
I think it can be useful to open a ticket on it and redefine every single implementation or getRequestUrl with a new one more specific (for example getAdminUrl), what do you think?

File: packages/core/admin/admin/src/components/AuthenticatedApp/utils/api.js
Comment (gu-stav) [inline packages/core/admin/admin/src/components/AuthenticatedApp/utils/api.js:41]: Should we only print these in development mode for now? Since we haven't started the migration this will print many messages in the console that might frighten our users.

File: packages/core/admin/admin/src/components/AuthenticatedApp/utils/api.js
Comment (gu-stav) [inline packages/core/admin/admin/src/components/AuthenticatedApp/utils/api.js:42]:
```suggestion
      'Deprecation warning: Usage of "axiosInstance" utility is deprecated and will be removed in the next major release. Instead, use the useFetchClient() hook, which is exported from the helper-plugin: { useFetchClient } from '@strapi/helper-plugin' '
```

cc @joshuaellis ?

File: packages/core/admin/admin/src/utils/fetchClient.js
Comment (gu-stav) [inline packages/core/admin/admin/src/utils/fetchClient.js:36]: Can we make this a named export and export this utility as `getFetchClient`? I'd also prefer to rename the file.

I'm wondering about something: would we export different instances of axios if we'd call this function multiple times? Should we use a singleton pattern instead to make sure `instance` and the interceptors are only generated once?


```js

export function getFetchClient() {
    if (!instance) {
      instance = axios.create();
      instance.interceptors.request.use();
    }
    
    return instance;
}
```

Also: can you please add tests for that utility? :) I'd personally test:

- we only exports HTTP verbs
- in case of a 401 response reload + clearing storage works
- headers are set properly

File: packages/core/admin/admin/src/pages/SettingsPage/pages/ApiTokens/EditView/index.js
Comment (gu-stav) [reply packages/core/admin/admin/src/pages/SettingsPage/pages/ApiTokens/EditView/index.js]: I agree with you - let's do this as a second step and clarify first, how we'd like to build URLs in the future :)

File: packages/core/helper-plugin/lib/src/hooks/useFetchClient/index.js
Comment (alexandrebodin) [inline packages/core/helper-plugin/lib/src/hooks/useFetchClient/index.js:4]: I don't know how often the StrapiAppContext  changes if ever but worth check if you need a memo in there

File: packages/core/admin/admin/src/utils/fetchClient.js
Comment (alexandrebodin) [reply packages/core/admin/admin/src/utils/fetchClient.js:36]: A nice approach (could be usefull for testing) is to export a default fetchClient & the getFetchClient utility as a proprty.

File: packages/core/helper-plugin/lib/src/hooks/useFetchClient/index.js
Comment (christiancp100) [reply packages/core/helper-plugin/lib/src/hooks/useFetchClient/index.js:4]: Do you mean using a memo like `useContext(useMemo(StrapiAppContext)` ?

Comment (alexandrebodin) [reply packages/core/helper-plugin/lib/src/hooks/useFetchClient/index.js:4]: I wouldn't do it this way but it doesn't matter if the StrapiAppContext isn't changing anyway you won't need it I was curious if you verified that :)

File: packages/core/admin/admin/src/core/utils/axiosInstance.js
Comment (alexandrebodin) [inline packages/core/admin/admin/src/core/utils/axiosInstance.js:4]: I'm not sure putting at the root of the modules is great as it will log on import so even before the functions are used. 

We could create a simple util to wrap the axiosInstance to log a warning on each call instead ?

Comment (gu-stav) [reply packages/core/admin/admin/src/core/utils/axiosInstance.js:4]: Yes, I agree. A central place for this warning would be nice, because for the start I'd like it be wrapped in a NODE_ENV=development check. As soon as we have good progress on the migration we can remove that but for now I'd like to avoid so many warnings at the same time.

File: packages/core/admin/admin/src/utils/fetchClient.js
Comment (gu-stav) [reply packages/core/admin/admin/src/utils/fetchClient.js:42]:
```js
import axios from 'axios';
import { auth } from '@strapi/helper-plugin';

let instance;

function getFetchClient() {
  if (instance) {
    return instance;
  }

  instance = axios.create({
    baseURL: process.env.STRAPI_ADMIN_BACKEND_URL,
  });

  instance.interceptors.request.use(
    async (config) => {
      config.headers = {
        Authorization: `Bearer ${auth.getToken()}`,
        Accept: 'application/json',
        'Content-Type': 'application/json',
      };

      return config;
    },
    (error) => {
      Promise.reject(error);
    }
  );

  instance.interceptors.response.use(
    (response) => response,
    (error) => {
      // whatever you want to do with the error
      if (error?.response?.status === 401) {
        auth.clearAppStorage();
        window.location.reload();
      }

      throw error;
    }
  );

  return instance;
}
```

export { getFetchClient }

File: packages/core/helper-plugin/lib/src/hooks/useFetchClient/index.js
Comment (gu-stav) [inline packages/core/helper-plugin/lib/src/hooks/useFetchClient/index.js:7]: Please only return the HTTP verbs here and not the whole instance ...

File: packages/core/admin/admin/src/utils/fetchClient.js
Comment (alexandrebodin) [reply packages/core/admin/admin/src/utils/fetchClient.js:42]: I would suggest this approach to benefit from a custom create for testing and mocking & the native singleton pattern that js modules provide.

```js
export const createFetchClient = () => { /* ... */ };

export default createFetchClient();
```

Comment (simotae14) [reply packages/core/admin/admin/src/utils/fetchClient.js:42]: ok, understood thanks @gu-stav

Comment (christiancp100) [reply packages/core/admin/admin/src/utils/fetchClient.js:42]: I might be wrong, but I don't see the need of re-implementing a singleton pattern here because ES6 modules will create a singleton instance by default, which [will be used in the whole code when imported](https://k94n.com/es6-modules-single-instance-pattern). 

Actually, we made the experiment, we created a hook that returned the Axios instance and instantiate it in one of our components, then instantiated the useFetchClient hook in the same component, and check if the instance exported by our custom hook and the one exported by our useFetchClient hook was the same, and it was. 

The only concern I see with this code now is that if we call the `getFetchClient`, yes, we would be creating another Axios instance, so following the explanation above, in my opinion, we could create the Axios instance outside the getFetchClient function and use this function just like a getter. However, I don't see a bad option to export just the methods and not the instance itself, because if we do so, it might be changed and it would be a change in the global scope of the application because of the singleton pattern (please correct me if I'm wrong somewhere 😅 ). 

So, instead of returning the instance in the getFetchClient I would return the methods to make the request:

```javascript
const instance = axios.create({
    baseURL: process.env.STRAPI_ADMIN_BACKEND_URL,
  });

  instance.interceptors.request.use(
    async (config) => {
      config.headers = {
        Authorization: `Bearer ${auth.getToken()}`,
        Accept: 'application/json',
        'Content-Type': 'application/json',
      };

      return config;
    },
    (error) => {
      Promise.reject(error);
    }
  );

  instance.interceptors.response.use(
    (response) => response,
    (error) => {
      // whatever you want to do with the error
      if (error?.response?.status === 401) {
        auth.clearAppStorage();
        window.location.reload();
      }

      throw error;
    }
  );

const getFetchClient = () => ({
  get: instance.get,
  put: instance.put,
  post: instance.post,
  delete: instance.delete,
  getFetchClient,
});

export default getFetchClient;
```

**Review by simotae14 — COMMENTED** [review]
ok

File: packages/core/admin/admin/src/utils/fetchClient.js
Comment (alexandrebodin) [reply packages/core/admin/admin/src/utils/fetchClient.js:42]: @christiancp100 when you dereference the methods from an object you lost the object context (a.k.a `this`) 

I'm not sure how axios function will respond to this. in general it is much better to use a Facade/Abstract pattern to avoid breaking axios internals that may require this to be bound correctly or you will need to at least do 

```js
get: instance.get.bind(instance)
...
```

File: packages/core/admin/admin/src/core/utils/axiosInstance.js
Comment (alexandrebodin) [reply packages/core/admin/admin/src/core/utils/axiosInstance.js:4]: So a wrapper you could build could be very simple like 

```js

const wrapAxiosDeprecationWarnings = (axiosInstance) => {


return {
	get(...arguments) {
		console.log('warning');
		return axiosInstance.get(...arguments)
	},
	......
  }


// or automated based on the doc here https://axios-http.com/docs/instance (might not be the matching version)

const wrapper = {};

['request', 'get', 'head', 'delete'', 'options', 'post', 'put', 'patch', 'getUri' ].forEach(methodName => {

wrapper[methodName] = (...args) => axiosInstance[methodName]
})

return wrapper
}

```

Comment (simotae14) [reply packages/core/admin/admin/src/core/utils/axiosInstance.js:4]: so basically I have to replace every call to the axiosInstance and instead call the wrapAxiosDeprecationWarnings?

Comment (alexandrebodin) [reply packages/core/admin/admin/src/core/utils/axiosInstance.js:4]: No, you will need to export wrapAxiosDeprecationWarnings(axiosIsntance) from the axiosInstance files only

File: packages/core/admin/admin/src/utils/fetchClient.js
Comment (christiancp100) [reply packages/core/admin/admin/src/utils/fetchClient.js:42]: Or if the goal is completely to decouple our implementation from Axios we could create a wrapper such as:
```javascript
const getFetchClient = () => ({
  get: (url, config) => instance.get(url, config),
  put: (url, data, config) => instance.put(url, data, config),
  ...
```

File: packages/core/admin/admin/src/core/utils/axiosInstance.js
Comment (simotae14) [reply packages/core/admin/admin/src/core/utils/axiosInstance.js:4]: ok thanks

File: packages/core/admin/admin/src/utils/fetchClient.js
Comment (alexandrebodin) [reply packages/core/admin/admin/src/utils/fetchClient.js:42]: Yes that's my suggestion (1st example)

File: packages/core/content-type-builder/admin/src/utils/axiosInstance.js
Comment (alexandrebodin) [inline packages/core/content-type-builder/admin/src/utils/axiosInstance.js:38]: You could put that in the helper plugin only once and export it to use instead of copy/pasting it probably

File: packages/core/admin/admin/src/utils/fetchClient.js
Comment (christiancp100) [reply packages/core/admin/admin/src/utils/fetchClient.js:42]: True 😅

File: packages/core/content-type-builder/admin/src/utils/axiosInstance.js
Comment (simotae14) [reply packages/core/content-type-builder/admin/src/utils/axiosInstance.js]: ok done, thanks

**Review by alexandrebodin — CHANGES_REQUESTED** [review]

File: packages/core/admin/admin/src/utils/fetchClient.js
Comment (simotae14) [reply packages/core/admin/admin/src/utils/fetchClient.js]: I have implemented yours suggestions

File: packages/core/admin/admin/src/utils/getFetchClient.js
Comment (gu-stav) [inline packages/core/admin/admin/src/utils/getFetchClient.js:34]: Could you also add a test for this interceptor?

File: packages/core/admin/admin/src/utils/getFetchClient.js
Comment (gu-stav) [inline packages/core/admin/admin/src/utils/getFetchClient.js:5]: Could you add a test case for this line?

File: packages/core/helper-plugin/lib/src/utils/wrapAxiosInstance/index.js
Comment (gu-stav) [inline packages/core/helper-plugin/lib/src/utils/wrapAxiosInstance/index.js:6]: I've asked this a couple of times already: can we please wrap this call in a `NODE_ENV==='development'` check for now?

File: packages/core/admin/admin/src/core/utils/axiosInstance.js
Comment (simotae14) [reply packages/core/admin/admin/src/core/utils/axiosInstance.js]: Sorry, I missed the previous request on it, I was not trying to upset you, sorry again...now it is done

File: packages/core/admin/admin/src/pages/SettingsPage/pages/Webhooks/EditView/index.js
Comment (simotae14) [reply packages/core/admin/admin/src/pages/SettingsPage/pages/Webhooks/EditView/index.js:67]: yes you are right, I added the warning and then I forgot to revert it back to the shorthand form

Comment (simotae14) [reply packages/core/admin/admin/src/pages/SettingsPage/pages/Webhooks/EditView/index.js]: done

File: packages/core/admin/admin/src/core/utils/axiosInstance.js
Comment (alexandrebodin) [inline packages/core/admin/admin/src/core/utils/axiosInstance.js:36]: should we just put that in the wrapper code instead ?

Comment (simotae14) [reply packages/core/admin/admin/src/core/utils/axiosInstance.js:36]: ok sure

File: packages/core/admin/admin/src/utils/getFetchClient.js
Comment (simotae14) [reply packages/core/admin/admin/src/utils/getFetchClient.js:34]: done

File: packages/core/admin/admin/src/utils/getFetchClient.js
Comment (christiancp100) [reply packages/core/admin/admin/src/utils/getFetchClient.js:5]: What do we want to test here exactly?

Comment (simotae14) [reply packages/core/admin/admin/src/utils/getFetchClient.js]: we have tried to test it, let me know if it is ok, thanks

**Review by joshuaellis — CHANGES_REQUESTED** [review]

File: packages/core/admin/admin/src/utils/getFetchClient.js
Comment (christiancp100) [reply packages/core/admin/admin/src/utils/getFetchClient.js:5]: Yes, our first option was to do that, it seemed unnecessary to us to pass the Axios instance through the provider when it doesn’t really have a state. Instead, we could just create a singleton function like this and call it through the hook directly. How do you see this? https://www.notion.so/strapi/ADRs-Architecture-Decision-Records-b5b9042c38f7449088cbaba540904361?p=a7d01bde64d44b1c84570983923dbc7b&pm=s

Comment (joshuaellis) [reply packages/core/admin/admin/src/utils/getFetchClient.js:5]: I would, why are we adding additional unnecessary complexity via out context providers, they're already tricky to manage.

Comment (christiancp100) [reply packages/core/admin/admin/src/utils/getFetchClient.js:5]: @alexandrebodin could you approve this?

Comment (alexandrebodin) [reply packages/core/admin/admin/src/utils/getFetchClient.js:5]: The only reason to pass it via providers IMO is if it needs to be initialized at the StrapiApp level. No problem doing this in the hook instead if that's sufficient 👍 

That being said looking at the code it looks like you don't even need a hook to access this 🤔  a simple import of the instance would work here. 

FYI reading this code I don't really see the point of using this singleton pattern like this. it makes only sense when you need to pass arguments for the 1st initialization or something. 

Here you can just define the instance in the module and export it as modules are singletons by nature.


```js
// export it for testing purposes and pass dependencies as params for testing too
export const createFetchClient = ({ baseURL }) => {
  return axios.create();
}

export default createFetchClient({ baseURL: process.env.STRAPI_ADMIN_BACKEND_URL })
```

It looks like you could just do  `import fechClient from '@strapi/admin'`

Comment (joshuaellis) [reply packages/core/admin/admin/src/utils/getFetchClient.js:5]: > Here you can just define the instance in the module and export it as modules are singletons by nature.

I like this. The _only_ reason I can think of writing a `useFetchClient` hook would be if we are going to implement a cancelling mechanism for if the component is unmounted before the `axios` request has finished. Whereby we can cancel calls when the hook is unmounted. Because the hook returns and adds `signal` to the calls only those calls made by the specific hook would be cancelled as opposed to all the calls the instance wants to make.

There are other scenarios we could manage e.g. avoiding race conditions on multiple requests but that might be out of scope for now and may be something axios handles already.

Comment (alexandrebodin) [reply packages/core/admin/admin/src/utils/getFetchClient.js:5]: That sounds good. We can mix the to and expose the imported instance through a hook to avoid breaking the api later on in that case.

Comment (simotae14) [reply packages/core/admin/admin/src/utils/getFetchClient.js:5]: Hello, thanks for your suggestions. The main idea before the implementation of this axios migration was to avoid duplicated implementations of the same utility. In addiction, we tried to avoid exposing it inside the helper-plugin because we know that its presence in the future is under discussion. I think these are the reasons to use the StrapiApp Context to pass the utility and to expose, in the beginning, the hook inside the helper plugin. Now your suggested solution means removing the getFetchClient from the StrapiApp, and having the implementation of the getFetchClient code inside the Admin and the hook inside the helper-plugin, I can also try to move the implementation of the getFetchClient code in the helper-plugin but in this case we are against our first idea, to add more things in the helper-plugin, wdyt?

Comment (alexandrebodin) [reply packages/core/admin/admin/src/utils/getFetchClient.js:5]: I'm not sure I see why we can't still do all this in `@strapi/admin` to plan for the helper plugin deprecation ?

Comment (simotae14) [reply packages/core/admin/admin/src/utils/getFetchClient.js:5]: yes I moved all the logic in the admin now

Comment (simotae14) [reply packages/core/admin/admin/src/utils/getFetchClient.js]: it is ready to be checked @joshuaellis and @alexandrebodin

File: packages/core/admin/admin/src/hooks/useFetchClient/index.js
Comment (joshuaellis) [inline packages/core/admin/admin/src/hooks/useFetchClient/index.js:6]: I think cancel token is deprecated – https://axios-http.com/docs/cancellation

File: packages/core/admin/admin/src/hooks/useFetchClient/index.js
Comment (joshuaellis) [inline packages/core/admin/admin/src/hooks/useFetchClient/index.js:13]: i'd store this in a ref so the value never changes, this would also mean you can omit the eslint warning you're suppressing.

Comment (joshuaellis) [reply packages/core/admin/admin/src/hooks/useFetchClient/index.js:13]:
```js
const source = useRef(null)

if(source.current === null) {
	source.current = cancelToken()
}

// ...
```

**Review by joshuaellis — CHANGES_REQUESTED** [review]
Small amends but I think it's nearly there ⭐

File: packages/core/admin/admin/src/hooks/useFetchClient/index.js
Comment (simotae14) [reply packages/core/admin/admin/src/hooks/useFetchClient/index.js]: Thanks a lot for your suggestion, @joshuaellis

**Review by christiancp100 — APPROVED** [review]
LGTM

**Review by alexandrebodin — DISMISSED** [review]
LGTM 👍  you might want to clean 1/2 comments that are left

**Review by joshuaellis — DISMISSED** [review]

**Comment by simotae14** [comment]
I have removed the places where I have used the new implementation to the old one and also some unnecessary comments, it is ready to be merged

File: packages/core/admin/admin/src/utils/tests/fetchClient.test.js
Comment (christiancp100) [inline packages/core/admin/admin/src/utils/tests/fetchClient.test.js:27]: Typo: function 😂

File: packages/core/admin/admin/src/utils/tests/fetchClient.test.js
Comment (christiancp100) [inline packages/core/admin/admin/src/utils/tests/fetchClient.test.js:69]: I would put this like: ``` `Bearer ${token}` ``` to reuse the defined variable

File: packages/core/admin/admin/src/utils/tests/getFetchClient.test.js
Comment (christiancp100) [inline packages/core/admin/admin/src/utils/tests/getFetchClient.test.js:5]: Is this necessary now?

File: packages/core/admin/admin/src/hooks/useFetchClient/index.js
Comment (joshuaellis) [inline packages/core/admin/admin/src/hooks/useFetchClient/index.js:4]: NIT, not sure if these needs to be it's own function?

**Review by joshuaellis — CHANGES_REQUESTED** [review]
minor tiny nit pick, @gu-stav should review this too.

File: packages/core/admin/admin/src/utils/tests/fetchClient.test.js
Comment (simotae14) [reply packages/core/admin/admin/src/utils/tests/fetchClient.test.js:27]: thanks

File: packages/core/admin/admin/src/utils/tests/fetchClient.test.js
Comment (simotae14) [reply packages/core/admin/admin/src/utils/tests/fetchClient.test.js:69]: yes, thanks

File: packages/core/admin/admin/src/utils/tests/getFetchClient.test.js
Comment (simotae14) [reply packages/core/admin/admin/src/utils/tests/getFetchClient.test.js:5]: yes, it is needed, otherwise a receive in the test `Bearer undefined`

Comment (simotae14) [reply packages/core/admin/admin/src/utils/tests/getFetchClient.test.js:5]: I have removed the mock of the auth.clearAppStorage

File: packages/core/admin/admin/src/utils/tests/fetchClient.test.js
Comment (simotae14) [reply packages/core/admin/admin/src/utils/tests/fetchClient.test.js:32]: ok done

File: packages/core/admin/admin/src/hooks/useFetchClient/index.js
Comment (simotae14) [reply packages/core/admin/admin/src/hooks/useFetchClient/index.js:4]: yes, I agree is just for two lines of code, I have removed the function and added directly the line in the if block

**Review by joshuaellis — APPROVED** [review]

**Review by christiancp100 — APPROVED** [review]
LGTM

**Comment by simotae14** [comment]
the PR seems ready to be merged, let me know @gu-stav if you are ok with the code, thanks

**Review by gu-stav — APPROVED** [review]

**Pull request merged**
